### ISSUE TITLE
Flutter and Firestore

### ISSUE THREAD

**Issue by HannHank** [flutter/flutter#42761]
Hello, I am very new to flutter and also not very familiar with Firestore. 
I am searching for a way to query in my database for a unique key. 

My Database looks like this:
 Tickets -->
                  g234hwjrw23 -->
                                             value: 10
                  h234jknkdjfkd --> 
                                             value: 8
I Just wanna now if a key like "g234hwjrw23" exists. So like a simple lookUp function. And I cannot find a code snipped, which do this for me

Thanks in advanced.

**Comment by HannHank** [comment]
Okay I now figured out how to do something like a Look up: 
`  var check = databaseReference
        .collection('Tickets')
        .document(qrText);
        print(check.path);
        print(check.get().then((snapshot){
        print(snapshot.data);
        if(snapshot.data != null){
        print("Ticket was found");
        }else{
        print("Ticket not found");
       }
    }));`
But the Problem is, that this is not working correctly because if I set the value 
` .document(qrText);`
with a variable, which is defined earlier, the Output is "null". So it means no Ticket was found. But this could not be true, because I log the path of the Ticket in that line:
`check.path`
And it show the correct path: "Ticket/g234hwjrw23" but it didn't work.
It just work if I set the path of the Ticket directly like that: 
`.document("g234hwjrw23");`
I don't understand that issue. In both setups the path of the Ticket looks the same in the Terminal. 
Help :)
